tests/Integration/Core/Configuration/AbstractMultistoreConfigurationTest.php

```
@@ -0,0 +1,168 @@
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+use Tests\Resources\DummyMultistoreConfiguration;
+
+class AbstractMultistoreConfigurationTest extends KernelTestCase
+{
+    /**
+     * @var Configuration
+     */
+    protected $legacyConfiguration;
+
+    /**
+     * @var Context
+     */
+    protected $shopContext;
+
+    /**
+     * @var FeatureInterface
+     */
+    protected $multistoreFeature;
+
+    /**
+     * @var Shop
+     */
+    protected $newShop;
+
+    public function setUp(): void
+    {
+        self::bootKernel();
+        $this->entityManager = self::$kernel->getContainer()->get('doctrine.orm.entity_manager');
+        $this->legacyConfiguration = self::$kernel->getContainer()->get('prestashop.adapter.legacy.configuration');
+        $this->initMultistore();
+        $this->multistoreFeature = self::$kernel->getContainer()->get('prestashop.adapter.multistore_feature');
+    }
+
+    /**
+     * @dataProvider provideShopConstraints
+     */
+    public function testUpdate(?ShopConstraint $shopConstraint): void
+    {
+        if (null === $shopConstraint) {
+            $shopConstraint = ShopConstraint::shop((int) $this->newShop->id);
+        }
+
+        // we mock the shop context so that its `getShopConstraint` method returns the ShopConstraint from our provider
+        $this->shopContext = $this->createShopContextMock();
+        $this->shopContext
+            ->method('getShopConstraint')
+            ->willReturn($shopConstraint);
+
+        $testedObject = new DummyMultistoreConfiguration(
+            $this->legacyConfiguration,
+            $this->shopContext,
+            $this->multistoreFeature
+        );
+
+        // test with multistore checkboxes, data should be saved for current context
+        $testedObject->updateConfiguration(['test_conf_1' => true, MultistoreCheckboxEnabler::MULTISTORE_FIELD_PREFIX . 'test_conf_1' => true]);
```

The first constraint is `ShopConstraint::allShops()` I thought in this case only the configuration fields are accepted, the resolver should throw an exception because some unexpected fields are present (the multi shop ones), no?